frontend/front/src/features/login/loginSlice.js

```diff
@@ -1,31 +1,50 @@
-import { createSlice } from "@reduxjs/toolkit";
+import {
+  AUTHORIZATION_HEADER,
+  AUTH_TOKEN_LOCAL_STORAGE_KEY,
+} from "./loginUtils";
+import { createSelector, createSlice } from "@reduxjs/toolkit";
 
-const initialState = {
-  userName: null,
-  password: null,
-  authenticated: true,
-};
+import { apiSlice } from "../../api/apiSlice";
 
 const loginSlice = createSlice({
   name: "login",
-  initialState,
+  initialState: { user: null, token: null },
   reducers: {
-    setLogin: (state, action) => {
-      state.userName = action.payload.userName;
-      state.password = action.payload.password;
-    },
-    setLogOut: (state) => {
-      state.userName = null;
-      state.password = null;
-    },
-    setAuthenticated: (state, action) => {
-      state.authenticated = action.payload;
+    setLoginInfo: (state, action) => {
+      state.user = action.payload.user;
+      state.token = action.payload.token;
     },
   },
+  extraReducers: (builder) => {
+    // update state whenever user logs in successfully
+    builder.addMatcher(
```

Comment:
This business automatically puts the `authorization` header in redux + localstorage on login